app/scripts/lib/channels/duplex.js

```diff
@@ -153,8 +166,11 @@ define(function (require, exports, module) {
      * @param {Object} [data]
      * @return {String}
      */
+    _messageCount: 0,
```

Comment:
It is, thanks for noticing! I went with a top level module variable instead of an instance variable because there can be more than one instance of a channel, and it's possible if both send a message in the same millisecond, they'd both have the same messageId. This avoids that.